src/Readers/ODataReader.v4/ODataCapabilitiesReader.cs

```diff
@@ -0,0 +1,196 @@
+            odcmProperty.Projection.EnsureCapabilities().Updateable = updatable;
+
+            foreach (var propertyPath in nonupdatableNavigationProperties)
+            {
+                OdcmProperty navProperty;
+                if (!(odcmProperty.Type as OdcmClass).TryFindProperty(propertyPath, out navProperty))
+                {
+                    throw new InvalidOperationException();
+                }
+                navProperty.Projection.EnsureCapabilities().Updateable = false;
+            }
+        }
+
+        private static void GetDeleteRestrictions(OdcmProperty odcmProperty, IEdmModel model, IEdmEntitySet entitySet)
+        {
+            IEdmValueAnnotation deleteAnnotation = FindVocabularyAnnotation(model, entitySet, s_deleteRestrictionsTerm);
+            if (deleteAnnotation == null)
+            {
+                return;
+            }
+
+            bool deletable;
+            IEnumerable<string> nonDeletableNavigationProperties;
+            GetBooleanAndPathCollection(deleteAnnotation, "Deletable", "NonDeletableNavigationProperties", out deletable, out nonDeletableNavigationProperties);
+            odcmProperty.Projection.EnsureCapabilities().Deleteable = deletable;
+
+            foreach (var propertyPath in nonDeletableNavigationProperties)
+            {
+                OdcmProperty navProperty;
+                if (!(odcmProperty.Type as OdcmClass).TryFindProperty(propertyPath, out navProperty))
+                {
+                    throw new InvalidOperationException();
+                }
+                navProperty.Projection.EnsureCapabilities().Deleteable = false;
+            }
+        }
+
+        private static void GetExpandRestrictions(OdcmProperty odcmProperty, IEdmModel model, IEdmEntitySet entitySet)
+        {
+            IEdmValueAnnotation expandAnnotation = FindVocabularyAnnotation(model, entitySet, s_expandRestrictionsTerm);
+            if (expandAnnotation == null)
+            {
+                return;
+            }
+
+            bool expandable;
+            IEnumerable<string> nonExpandableProperties;
+            GetBooleanAndPathCollection(expandAnnotation, "Expandable", "NonExpandableProperties", out expandable, out nonExpandableProperties);
+            odcmProperty.Projection.EnsureCapabilities().Expandable = expandable;
+
+            foreach (var propertyPath in nonExpandableProperties)
+            {
+                OdcmProperty navProperty;
+                if (!(odcmProperty.Type as OdcmClass).TryFindProperty(propertyPath, out navProperty))
+                {
+                    throw new InvalidOperationException();
+                }
+                navProperty.Projection.EnsureCapabilities().Expandable = false;
+            }
+        }
+
+        private static void GetBooleanAndPathCollection(IEdmValueAnnotation annotation, string booleanPropertyName, string pathsPropertyName, out bool boolean, out IEnumerable<string> paths)
```

Comment:
Can we use the return value instead of an out parameter? Out parameters are generally used in TrySomething(out result) scenarios.